account_followup/models/res_partner.py

```diff
@@ -0,0 +1,447 @@
+                raise ValidationError(
+                    _("There is no followup plan defined for the current "
+                      "company."))
+            data = {
+                'date': fields.date.today(),
+                'followup_id': followup_ids[0],
+            }
+            # call the print overdue report on this partner
+            return self.do_partner_print(wizard_partner_ids, data)
+
+    @api.multi
+    @api.depends('payment_amount_due', 'payment_amount_overdue',
+                 'payment_earliest_due_date')
+    def _compute_amounts_and_date(self):
+        """ Function that computes values for the followup functional fields.
+            Note that 'payment_amount_due' is similar to 'credit' field on
+            res.partner except it filters on user's company.
+        """
+        res = {}
+        company = self.env.user.company_id
+        current_date = fields.Date.context_today(self)
+        for partner in self:
+            worst_due_date = False
+            amount_due = amount_overdue = 0.0
+            for aml in partner.unreconciled_aml_ids:
+                if aml.company_id == company:
+                    date_maturity = aml.date_maturity or aml.date
+                    if not worst_due_date or date_maturity < worst_due_date:
+                        worst_due_date = date_maturity
+                    amount_due += aml.result
+                    if date_maturity <= current_date:
+                        amount_overdue += aml.result
+            res[partner.id] = {
+                'payment_amount_due': amount_due,
+                'payment_amount_overdue': amount_overdue,
+                'payment_earliest_due_date': worst_due_date
+            }
+        return res
+
+    @api.model
+    def _get_followup_overdue_query(self, args, overdue_only=False):
+        """
+        This function is used to build the query and arguments to use when
+        making a search on functional fields
+            * payment_amount_due
+            * payment_amount_overdue
+        Basically, the query is exactly the same except that for overdue there
+        is an extra clause in the WHERE.
+
+        :param args: arguments given to the search in the usual domain
+            notation (list of tuples)
+        :param overdue_only: option to add the extra argument to filter on
+            overdue accounting entries or not
+        :returns: a tuple with
+            * the query to execute as first element
+            * the arguments for the execution of this query
+        :rtype: (string, [])
+        """
+        company_id = self.env.user.company_id.id
+        having_where_clause = ' AND '.join(
+            map(lambda x: '(SUM(bal2) %s %%s)' % (x[1]), args))
+        having_values = [x[2] for x in args]
+        query = self.env['account.move.line']._query_get()
+        overdue_only_str = overdue_only and 'AND date_maturity <= NOW()' or ''
+        return ("""
+            SELECT pid AS partner_id, SUM(bal2)
```

Comment:
I don't like SQL very much, but we can live with this if there is performance problem. Have you tried in ORM?